services/repository/src/main/java/org/sagebionetworks/auth/filter/OAuthClientAuthFilter.java

```diff
@@ -1,91 +1,128 @@
-			HttpServletRequest httpRequest, HttpServletResponse httpResponse, 
-			FilterChain filterChain, Optional<UserNameAndPassword> credentials) throws IOException, ServletException {
-		if (credentials.isPresent() && !validCredentials(credentials.get())) {
-			rejectRequest(httpResponse, getInvalidCredentialsMessage());
-			return;
-		}
-
-		doFilterInternal(httpRequest, httpResponse, filterChain, credentials.orElse(null));
-	}
-
-	private boolean validCredentials(UserNameAndPassword credentials) {
-		OAuthClientIdAndSecret clientCreds = new OAuthClientIdAndSecret();
-
-		clientCreds.setClient_id(credentials.getUserName());
-		clientCreds.setClient_secret(credentials.getPassword());
-
-		return oauthClientManager.validateClientCredentials(clientCreds);
-	}
-
-	private void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain, UserNameAndPassword credentials) throws ServletException, IOException {
-		
-		if (credentials == null) {
-			throw new IllegalStateException("Credentials were expected but not supplied");
-		}
-
-		String oauthClientId = credentials.getUserName();
-
-		// get the current headers, but be sure to leave behind anything that might be
-		// mistaken for a valid
-		// authentication header 'down the filter chain'
-
-		Map<String, String[]> modHeaders = HttpAuthUtil.filterAuthorizationHeaders(request);
-		modHeaders.put(AuthorizationConstants.OAUTH_VERIFIED_CLIENT_ID_HEADER, new String[] { oauthClientId });
-		HttpServletRequest modRqst = new ModHttpServletRequest(request, modHeaders, null);
-
-		filterChain.doFilter(modRqst, response);
-	}
-
-}
+package org.sagebionetworks.auth.filter;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sagebionetworks.StackConfiguration;
+import org.sagebionetworks.auth.HttpAuthUtil;
+import org.sagebionetworks.auth.UserNameAndPassword;
+import org.sagebionetworks.authutil.ModHttpServletRequest;
+import org.sagebionetworks.cloudwatch.Consumer;
+import org.sagebionetworks.repo.manager.oauth.OAuthClientManager;
+import org.sagebionetworks.repo.model.AuthorizationConstants;
+import org.sagebionetworks.repo.model.oauth.OAuthClientIdAndSecret;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implements the client_secret_basic and client_secret_post protocols for authenticating an OAuth client
+ * See: https://tools.ietf.org/html/rfc6749#section-2.3.1
+ */
+@Component("oauthClientAuthFilter")
+public class OAuthClientAuthFilter implements Filter {

```

Comment:
Might be worth leaving it as a basic auth filter and add additional logic to try the alternative